hyperspy/tests/signal/test_tools.py

```diff
@@ -184,6 +185,42 @@ def test_numpy_func(self):
         assert isinstance(result, np.ndarray)
         np.testing.assert_array_equal(result, np.angle(self.signal.data))
 
+    def test_add_gaussian_noise(self):
+        s = self.signal
+        kwargs = {}
+        if s._lazy:
+            data = s.data.compute()
+            from dask.array.random import seed, normal
+            kwargs["chunks"] = s.data.chunks
+        else:
+            data = s.data.copy()
+            from numpy.random import seed, normal
+        seed(1)
+        s.add_gaussian_noise(std=1.0)
+        seed(1)
+        if s._lazy:
+            s.compute()
+        np.testing.assert_array_almost_equal(
```

Comment:
If you do `s.compute` for lazy signal, then `s` become a non lazy signal and in the tests there is no need to use the `normal` function from dask?

Reply:
I also thought so, but it turns out that the random numbers generated by dask and numpy with the same seed differ.

Reply:
The tests are failing with the same systematic error. Could it be because of the seed issue?